vscode.sh

```diff
@@ -68,3 +68,5 @@ done
 zenity --error --no-wrap --title='Visual Studio Code is not installed' \
   --text="Visual Studio Code is required to edit scripts with the Unity Hub Flatpak, please \
 install it from Flathub."
+gdbus call -e -d org.gnome.Software -o /org/gnome/Software/SearchProvider  \
```

Comment:
You can probably use the GtkApplication actions to do the equivalent of `gnome-software --details=...`. In D-Bus monitor I see this:
```
method call time=1552037382.773304 sender=:1.1262 -> destination=org.gnome.Software serial=5 path=/org/gnome/Software; interface=org.gtk.Actions; member=Activate
   string "details"
   array [
      variant          struct {
            string "com.visualstudio.code"
            string ""
         }
   ]
   array [
      dict entry(
         string "cwd"
         variant             array of bytes "/home/ramcq" + \0
      )
   ]
```

Is it possible to use Zenity in a prompting mode, so you can ask the user whether they would like to install, and then launch Software in that case, or exit 1 if they say no?

Reply:
@ramcq I'll put Zenity in prompting mode. I can change the software call if you want, but one of the advantages of the SearchProvider route is that the user can pick between VS Code proprietary or OSS. Then again, I'm not sure how much that would matter given that Unity Hub itself is proprietary and probably has telemetry in it (some people have an odd double standard with MS though).

Reply:
I got the search view in G-S from the above too, because it's an ambiguous app ID match.

Reply:
Well after like 30 minutes of debugging, turns out [details is broken in Software for Flatpaks at the moment](https://gitlab.gnome.org/GNOME/gnome-software/issues/577).